osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSliderRepeat.cs

```diff
@@ -85,8 +87,16 @@ protected override void OnApply()
 
         protected override void CheckForResult(bool userTriggered, double timeOffset)
         {
-            if (HitObject.StartTime <= Time.Current)
-                ApplyResult(r => r.Type = DrawableSlider.Tracking.Value ? r.Judgement.MaxResult : r.Judgement.MinResult);
+            // shared implementation with DrawableSliderTick.
+            if (timeOffset >= 0)
+            {
+                // Attempt to preserve correct ordering of judgements as best we can by forcing
+                // an un-judged head to be missed when the user has clearly skipped it.
+                if (Tracking && !DrawableSlider.HeadCircle.Judged)
+                    DrawableSlider.HeadCircle.MissForcefully();
```

Comment:
This repeated code can probably be extracted out if this approach is seen as a good direction, but I'm more interested in feedback on whether this makes sense first.

Reply:
Yeah I'm not sure either. I've clarified this in comments as best I can.